client/state/form/actions.js

```diff
@@ -7,3 +14,68 @@ export const setFormProperty = ( field, value ) => {
 		value,
 	};
 };
+
+export const goToStep = ( stepIndex ) => {
+	return {
+		type: GO_TO_STEP,
+		step: stepIndex,
+	};
+};
+
+export const nextStep = () => ( dispatch, getState, { callbackURL, nonce, formSchema, formLayout } ) => {
```

Comment:
Wait, does this just yield another function? If so.. why have `nextStep()` generate a function if it's not dependent on an initial set of arguments?

Can you explain this block a bit more?